### PR TITLE
Adds a way to bootstrapping the Chaos Monkey.

### PR DESCRIPTION
**Pull request by ncrash**
**What**: Adds a way to bootstrapping the Chaos Monkey.

**Why**: some project support only single profile. in this case we can't applied CM4SB. i want to resolve this restriction.

**How**: so that it can be activated as an system property with `@conditional` annotation instead of a spring active profile.

**Checklist**:

- [x] Documentation added
      
- [x] Changelog updated
      
- [x] Tests added
      
- [x] Ready to be merged

### CODE REVIEW

**Comment by ncrash** [comment]
@fletchgqc Could you review it? Please review it if possible.

**Comment by ncrash** [comment]
> Hi @ncrash
> 
> Thank you for your first contribution! 👍
> However, I am unsure if its a necessary addition.
> 
> In the past I also used a property called `spring.profiles.include` to include chaos monkey.
> 
> Then you could do something like this:
> `java -jar app.jar --spring.profiles.include=chaos-monkey`
> or supplied as a system property:
> `java -Dspring.profiles.include=chaos-monkey -jar app.jar`
> 
> Does this work for you? Or can you please elaborate a bit more about your use case.

Hi @WtfJoke 
thank you for reviewing this PR.

"spring.profiles.include" property setting you suggested doesn't solve my problem.
Because it is impossible to specify other than 3 profiles (local, develop, and production).
I requested this PR review because it is impossible to resolve this restriction.
I hope to use CM4SB even in environments where these restrictions exist.

**Comment by fletchgqc** [comment]
> Because it is impossible to specify other than 3 profiles (local, develop, and production).

Are you sure? I have successfully done it. Please try on a simple project where CMSB is not involved, and read the Spring log output during startup, it will tell you what is active.

**Comment by ncrash** [comment]
> > Because it is impossible to specify other than 3 profiles (local, develop, and production).
> 
> Are you sure? I have successfully done it. Please try on a simple project where CMSB is not involved, and read the Spring log output during startup, it will tell you what is active.

Hello, @fletchgqc.
Thank you for reviewing my PR.

I made a [simple project](https://github.com/ncrash/chaos-monkey-spring-boot-profile-issue) so that I can share my problems effectively through your advice. 

look at this code, you can see that only three profiles(local, develop, production) are allowed.
chaos-monkey profile could not be specified due to this code.
https://github.com/ncrash/chaos-monkey-spring-boot-profile-issue/commit/d5ab1c60481ec72ec86548b4873ed57bb1c7e56b#diff-8d916551cb4769e613d9b2b1647e4abf8d36cc45b9a0a5a2bd530f2eafa982feR3-R25

The reason why I proposed this PR is that the profile check logic exists in several modules that cannot be controlled.
I would like to ask you to positively review the PR so that we can use CM4SB in this situation.

**Comment by ncrash** [comment]
```
❯ java -jar build/libs/chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar --spring.profiles.active=local --spring.profiles.include=chaos-monkey

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.3.8.RELEASE)

2021-02-05 13:32:09.861  INFO 32898 --- [           main] com.example.demo.DemoApplication         : Starting DemoApplication on gangdaegwons-Mac-mini.local with PID 32898 (/Users/ncrash/Code/mine/chaos-monkey-spring-boot-profile-issue/build/libs/chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar started by ncrash in /Users/ncrash/Code/mine/chaos-monkey-spring-boot-profile-issue)
2021-02-05 13:32:09.868  INFO 32898 --- [           main] com.example.demo.DemoApplication         : The following profiles are active: chaos-monkey,local
2021-02-05 13:32:12.191  INFO 32898 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2021-02-05 13:32:12.223  INFO 32898 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2021-02-05 13:32:12.223  INFO 32898 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.41]
2021-02-05 13:32:12.415  INFO 32898 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2021-02-05 13:32:12.416  INFO 32898 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 2396 ms
2021-02-05 13:32:12.534  WARN 32898 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'profileFinder' defined in URL [jar:file:/Users/ncrash/Code/mine/chaos-monkey-spring-boot-profile-issue/build/libs/chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/com/example/demo/profile/ProfileFinder.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.example.demo.profile.ProfileFinder]: Constructor threw exception; nested exception is java.lang.IllegalStateException: must have at only one profile value.
2021-02-05 13:32:12.539  INFO 32898 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2021-02-05 13:32:12.563  INFO 32898 --- [           main] ConditionEvaluationReportLoggingListener :

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-02-05 13:32:12.580 ERROR 32898 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'profileFinder' defined in URL [jar:file:/Users/ncrash/Code/mine/chaos-monkey-spring-boot-profile-issue/build/libs/chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/com/example/demo/profile/ProfileFinder.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.example.demo.profile.ProfileFinder]: Constructor threw exception; nested exception is java.lang.IllegalStateException: must have at only one profile value.
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:314) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:295) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1356) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1203) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:556) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:516) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:324) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:897) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:879) ~[spring-context-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551) ~[spring-context-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:143) ~[spring-boot-2.3.8.RELEASE.jar!/:2.3.8.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:758) ~[spring-boot-2.3.8.RELEASE.jar!/:2.3.8.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:750) ~[spring-boot-2.3.8.RELEASE.jar!/:2.3.8.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:405) ~[spring-boot-2.3.8.RELEASE.jar!/:2.3.8.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.3.8.RELEASE.jar!/:2.3.8.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1237) ~[spring-boot-2.3.8.RELEASE.jar!/:2.3.8.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.3.8.RELEASE.jar!/:2.3.8.RELEASE]
	at com.example.demo.DemoApplication.main(DemoApplication.java:10) ~[classes!/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) ~[na:na]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:107) ~[chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) ~[chaos-monkey-spring-boot-profile-issue-0.0.1-SNAPSHOT.jar:na]
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.example.demo.profile.ProfileFinder]: Constructor threw exception; nested exception is java.lang.IllegalStateException: must have at only one profile value.
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:217) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:117) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:310) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	... 28 common frames omitted
Caused by: java.lang.IllegalStateException: must have at only one profile value.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:508) ~[guava-30.1-jre.jar!/:na]
	at com.example.demo.profile.ProfileFinder.<init>(ProfileFinder.java:17) ~[classes!/:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:204) ~[spring-beans-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	... 30 common frames omitted
```

**Comment by ncrash** [comment]
@fletchgqc @WtfJoke 
Could you please review it again?
I'd appreciate it if you could let me know if there's anything to improve

File: chaos-monkey-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -23,7 +23,11 @@ There are two ways to integrate the Chaos Monkey for Spring Boot into your exist
 [source,txt,subs="verbatim,attributes"]
 .command-line
 ----
+# spring profile
 java -jar your-app.jar --spring.profiles.active=chaos-monkey
+
+# system property
+java -DCHAOS_MONKEY=enabled -jar your-app.jar
```
Comment (WtfJoke) [inline chaos-monkey-docs/src/main/asciidoc/getting-started.adoc:30]:
```suggestion
java -jar your-app.jar -DCHAOS_MONKEY=enabled
```

**Review by WtfJoke — COMMENTED** [review]
Hi @ncrash 

Sorry for the long wait. I've gone through once more and thought about your change once again.

I made a small remark in the docs beside that I think we could merge this as soon as you bring your branch up to date and add the feature and yourself to the changelog

File: chaos-monkey-docs/src/main/asciidoc/getting-started.adoc (hunk 1 above)
Comment (WtfJoke) [reply chaos-monkey-docs/src/main/asciidoc/getting-started.adoc:30]: I think the property name LOAD_CHAOS_MONKEY=true would suit better

Comment (ncrash) [reply chaos-monkey-docs/src/main/asciidoc/getting-started.adoc]: Hi @WtfJoke 
thank you for reviewing my request.
i think the naming you suggested helps us improve code readability.
so I immediately fixed it in the code.

Comment (WtfJoke) [reply chaos-monkey-docs/src/main/asciidoc/getting-started.adoc]: Thanks!

**Review by WtfJoke — COMMENTED** [review]
Now only the [changelog entry](https://github.com/codecentric/chaos-monkey-spring-boot/blob/main/chaos-monkey-docs/src/main/asciidoc/changes.adoc) is missing. Else this looks good and can be merged.  👍 

Optional bonus if you want to make us really happy you can also add a JUnit Test for ChaosMonkeyCondition

**Comment by ncrash** [comment]
> Now only the [changelog entry](https://github.com/codecentric/chaos-monkey-spring-boot/blob/main/chaos-monkey-docs/src/main/asciidoc/changes.adoc) is missing. Else this looks good and can be merged. 👍
> 
> Optional bonus if you want to make us really happy you can also add a JUnit Test for ChaosMonkeyCondition

updated changelog and thanks to you I got a bonus :)

**Review by WtfJoke — APPROVED** [review]
LGTM 🚀

**Comment by WtfJoke** [comment]
I just realized that the pipeline is red. I think you need to run `./mvnw spotless:apply`

**Comment by checketts** [comment]
It is more common in Spring Boot to put 'enabled' in the property to denote both that the expected property is a boolean and to be consistent. See https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html#application-properties.data.spring.data.elasticsearch.repositories.enabled

I would recommend 'CHAOS_MONKEY_ENABLED'

**Comment by WtfJoke** [comment]
> I would recommend 'CHAOS_MONKEY_ENABLED'

I normally would agree with this. However we have two different settings:

```
spring.profiles.active=chaos-monkey
chaos.monkey.enabled=true
```

The spring.profile for loading chaos monkey configuration. And the chaos.monkey.enabled flag for disabling/enabling chaos monkey when chaos monkey configuration is loaded.

This newly introduced flag is an alternative for activating the spring.profile. And I fear two things if we would name it that way:
1. Confuse people which might want to use this environment variable which they then need to set CHAOS_MONKEY_ENABLED and the property chaos.monkey.enabled. So they might dont realize they need two do both of that. 
2. Introduce a clash, since the property chaos.monkey.enabled could be activated with CHAOS_MONKEY_ENABLED. So you would controll two things with one environment variable.


Neverthless im open for alternative name suggestions, I thought `LOAD_CHAOS_MONKEY` would make it clearer that its different from enabled (which you still need to set).

**Comment by checketts** [comment]
Oh I hadn't realized the difference. Now it make sense, we definitely won't want to use CHAOS_MONKEY_ENABLED in that case.  Perhaps CHAOS_MONKEY_LOADED would be another candidate?

I didn't mean to hop in an distract the PR. Carry on. ;)

**Comment by WtfJoke** [comment]
> Perhaps CHAOS_MONKEY_LOADED would be another candidate?

Yeah would be another candidate. It would match better from the pattern but loaded as a past term could be improved.
So I guess I'll leave it to the current one. 

> I didn't mean to hop in an distract the PR. Carry on. ;)

No worries, I appreciate your feedback :)

**Review by WtfJoke — APPROVED** [review]

**Pull request merged**